client/client.go

```diff
@@ -892,9 +948,6 @@ func (c *Client) registerNode() error {
 	}
 	var resp structs.NodeUpdateResponse
 	if err := c.RPC("Node.Register", &req, &resp); err != nil {
-		if time.Since(c.start) > registerErrGrace {
```

Comment:
It merely wrapped an error with a tiny bit more context in certain circumstances.

Instead I just added that tiny bit more context to the log line that handles this error, so there's no need to wrap it: https://github.com/hashicorp/nomad/pull/1735/files/ec81fc9baf776efb30bd752f64688ad562d915c6#diff-1cd41a2611da49ce63630d5a51bd82c5L876

It also kind of seemed like dead code since the error wrapping only affected logging. It seemed like maybe once upon a time it affected retry logic, but it doesn't  anymore on master or this PR.